src/control/common/cmdutil/logging.go

```diff
@@ -24,9 +28,82 @@ type (
 	LogCmd struct {
 		logging.Logger
 	}
+
+	// LogConfig contains parameters used to configure the logger.
+	LogConfig struct {
+		LogFile  string
+		LogLevel common.ControlLogLevel
+		JSON     bool
+	}
 )
 
 // SetLog sets the logger for the command.
 func (cmd *LogCmd) SetLog(log logging.Logger) {
 	cmd.Logger = log
 }
+
+// ConfigureLogger configures the logger according to the requested config.
+func ConfigureLogger(logIn logging.Logger, cfg LogConfig) error {
+	log, ok := logIn.(*logging.LeveledLogger)
```

Comment:
Is the typecast needed to get at the `.WithJSONOutput()` method?

Reply:
Yes, and the `.SetLevel()` method as well.